### ISSUE TITLE
Parameter/Function expressions not evaluated when running JMeter test with Gradle plugin

### ISSUE THREAD

**Issue by matthiasbasler** [jmeter-gradle-plugin/jmeter-gradle-plugin#115]
I have a Jeter test which is working well from the command line.
This test uses parameters in the form "${__P(PORT,1234)}". Using the command line script they can be passed either as -JPORT=1234 or specifying a jmeter.properties.

Now, when I run this very same test with the Gradle plugin, the above expressions are **not evaluated at all**, that is, I get for instance an invalid URL with "${__P(PORT,1234)}" built in it. If it was only a problem with passing parameter values to JMeter I would expect the default value to be used, but this is not the case. More strange, evaluating "normal" variables works, so if I set the port to ${port} and assign "1234" to the "port" variable, then the variable is evaluated and the test runs.

I wonder if I have some misconfiguration or if there is a bug in the current version 1.0.10-3.3 of the plugin? (I can hardly imagine so.)

My gradle configuration is:
_____
buildscript {
	repositories {
		jcenter()
		mavenCentral()
		maven {
		  url "https://plugins.gradle.org/m2/"
		}
	}
}

plugins {
  id "net.foragerr.jmeter" version "1.0.10-3.3"
}
...
jmeter {
	jmTestFiles = [file("src/test/jmeter/PropertyTest.jmx")] // run a test
	jmSystemPropertiesFiles= [file("src/test/jmeter/jmeter.properties")] 
}

//See https://github.com/jmeter-gradle-plugin/jmeter-gradle-plugin/issues/109
task copyToBin(type: Copy) {
    from "$projectDir/src/main/resources"
    into "$buildDir/jmeter/bin"
    include "log4j2.xml"
		printf 'log4j2.xml kopiert.\n'
}

jmRun.dependsOn copyToBin
jmGui.dependsOn copyToBin
jmeter {
	//testFileDir = file("src/test/jmeter") // Path to files. Default is src/test/jmeter
	jmTestFiles = [file("src/test/jmeter/PropertyTest.jmx")] //if jmx file is not in the default location
	jmSystemPropertiesFiles= [file("src/test/jmeter/jmeter.properties")] //to add additional system properties
	//jmUserProperties = ["protocol=HTTPS", "server=gisx3server", "port=8444"]
}

//Hack for current version oj JMeter "1.0.10-3.3"
//See https://github.com/jmeter-gradle-plugin/jmeter-gradle-plugin/issues/109
task copyToBin(type: Copy) {
    from "$projectDir/src/main/resources"
    into "$buildDir/jmeter/bin"
    include "log4j2.xml"
		printf 'log4j2.xml kopiert.\n'
}

jmRun.dependsOn copyToBin
jmGui.dependsOn copyToBin
`
_____

**Any idea what could cause this behaviour of not evaluating function/parameter expressions?**

Side note: When running the gradle task I get several warnings of the form 
_____
[INFO ] 02.03.2018 08:56:28 [main] JMeter - search_paths=/C:/Users/<user>/.gradle/caches/modules-2/files-2.1/org.apache.jmeter/ApacheJMeter_junit/3.3/195173bfdd03b6fdd5624dc5cfb3d52a65785c4b/ApacheJMeter_junit-3.3.jar;
/C:/Users/<user>/.gradle/caches/modules-2/files-2.1/org.apache.jmeter/ApacheJMeter_mail/3.3/2b982a66df3e30fa0cefe655212068635f119e40/ApacheJMeter_mail-3.3.jar; ...
[WARN ] 02.03.2018 08:56:28 [main] JMeter - Can't read /C:/Users/<user>/.gradle/caches/modules-2/files-2.1/org.apache.jmeter/ApacheJMeter_junit/3.3/195173bfdd03b6fdd5624dc5cfb3d52a65785c4b/ApacheJMeter_junit-3.3.jar
[WARN ] 02.03.2018 08:56:28 [main] JMeter - Can't read /C:/Users/<user>/.gradle/caches/modules-2/files-2.1/org.apache.jmeter/ApacheJMeter_mail/3.3/2b982a66df3e30fa0cefe655212068635f119e40/ApacheJMeter_mail-3.3.jar
...
_____
The files listed as "Can't read" do actually exist, so no idea what the problem is.
Not sure if this is related to the above issue ... otherwise I would simply ignore the warnings.

System: Win7 on Eclipse 4.7.2.

**Comment by vladimirPe** [comment]
Check the path to the Gradle cache directory (.gradle) it must not have the spaces. To solve I've used
`gradle -g /some/other/dir jmRun`

**Comment by matthiasbasler** [comment]
You are right, my user name has a space in it, so my gradle cache is located under
"/C:/Users/My Name/.gradle" (only an example here).
Interestingly Gradle works fine so far with this cache location, except for this plugin. So I am a bit reluctant to move the gradle cache somewhere else. Maybe the plugin can be improved to cope with such paths?
Anyway, thanks for pointing me to the source of the problem.

P.S. We have used batch files instead to run the JMeter tests as a workaround.
